.github/workflows/e2e.yml

```diff
@@ -39,7 +38,37 @@ jobs:
       || (contains(github.event.pull_request.labels.*.name, 'build')
          && (github.event.action != 'labeled' || github.event.label.name == 'build')
          )
+    steps:
+      - uses: actions/checkout@v2.3.4
+        with:
+          path: plugin
+      - uses: dorny/paths-filter@b2feaf19c27470162a626bd6fa8438ae5b263721
+        if: github.event_name != 'workflow_dispatch' && github.event_name != 'push'
+        id: filter
+        with:
+          working-directory: plugin
+          filters: |
+            e2e-test:
+              - '**/e2e-test/**'
+      - name: Set e2e-tests list
+        id: spl
+        run: |
+          if [ ${{ github.event_name }} != 'workflow_dispatch' ] && [ ${{ github.event_name }} != 'push' ] && [ ${{ steps.filter.outputs.e2e-test }} == 'false' ]; then
```

Comment:
suppose there is no change in e2e-tests in the PR, it will end up running every tests, right?

Reply:
No. `steps.filter.outputs.e2e-test` will be `false` when there is no change in e2e-tests so only required tests will be run.